modules/avformat/avformat.c

```diff
@@ -203,7 +204,13 @@ int avformat_shared_alloc(struct shared **shp, const char *dev)
 	if (err)
 		goto out;
 
-	ret = avformat_open_input(&st->ic, dev, NULL, NULL);
+	ff_const59 AVInputFormat *input_format;
```

Comment:
move declaration to the top of the function.

is `ff_const59 ` needed ? can we use `const` instead ?